base/sparse/sparsevector.jl

```diff
@@ -1518,6 +1536,19 @@ end
 
 ### BLAS-2 / dense A * sparse x -> dense y
 
+# lowrankupdate (BLAS.ger! like)
+function LinAlg.lowrankupdate!(A::StridedMatrix, x::StridedVector, y::SparseVectorUnion, α::Number = 1)
```

Comment:
broadcast, and any generic dense implementations, would handle the corner cases here. We don't need to repeat other libraries' bugs when it's a fairly straightforward (and not that expensive) check to do the slow but right thing in rare cases, instead of pretending the cases don't exist and ignoring them.